File: server/src/main/java/com/smartexpense/api/application/dto/response/UserProfileResponseDto.java
package com.smartexpense.api.application.dto.response;

import lombok.Builder;
import lombok.Data;
import java.math.BigDecimal;
import java.util.UUID;

@Data
@Builder
public class UserProfileResponseDto {
    private UUID id;
    private String email;
    private String displayName;
    private String avatarUrl;
    private BigDecimal monthlyBudget;
    private Integer streakDays;
    private String themeColor;
}

File: server/src/main/kotlin/com/smartexpense/api/application/dto/response/UserProfileResponseDto.kt
package com.smartexpense.api.application.dto.response

import java.math.BigDecimal
import java.util.UUID

data class UserProfileResponseDto(
    val id: UUID,
    val email: String,
    val displayName: String,
    val avatarUrl: String?,
    val monthlyBudget: BigDecimal?,
    val streakDays: Int,
    val themeColor: String
)
